src/darwin/Framework/CHIP/CHIPPersistentStorageDelegateBridge.mm

```diff
@@ -120,17 +120,21 @@
         }
 
         if (valueString != nil) {
-            if (value != nullptr) {
-                size = strlcpy(value, [valueString UTF8String], size);
-                if (size < [valueString length]) {
+            if (([valueString length] + 1) > UINT16_MAX) {
+                error = CHIP_ERROR_BUFFER_TOO_SMALL;
+            } else {
+                if (value != nullptr) {
+                    size = (uint16_t) strlcpy(value, [valueString UTF8String], size);
```

Comment:
Is this right?  I would think `[valueString length]` is the number of 16-bit UTF-16 units, which tells you very little about the length of the UTF8String representation, unless we know the string is ASCII.

I guess we already had this issue with the existing `size < [valueString length]` check?

Reply:
Maybe we should use `[valueString lengthOfBytesUsingEncoding:NSUTF8StringEncoding]`?